### PR TITLE
{Network} `az network dns record-set naptr`: Naptr feature tests and fixes

### PR DESCRIPTION
**Pull request by WhiteHorseV**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by WhiteHorseV** [comment]
More changes to follow after the NAPTR feature is enabled in PPE/Prod when I can test the working. I will record the expected results after the feature is enabled.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py [hunk 1]
```diff
@@ -575,6 +588,10 @@ def test_zone_file_1(self):
             (60, 0, 'issue', 'ca1.contoso.com'),
             (60, 45, 'tag56', 'test test test')
         ])
+        #self._check_naptr(zone, 'mynaptr.' + zn, [
+        #    (3600, 10, 20, 'A', 'EAU+SIP', '', 'domain.com.'),
+        #    (3600, 20, 20, 'U', 'SIP+D2U', '!^(\\+441632960083)$!sip:\\1@example.com!', '.')
+        #])
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py:594]: why is this commented?

File: src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py [hunk 2]
```diff
@@ -207,6 +208,8 @@ def _serialize(tokens):
     for tok in tokens:
         if tok is None:
             continue
+        elif tok == '':
+            tok = 'EMPTY'
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py:212]: can this "EMPTY" affect other records?

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/zone_files/zone10.txt [hunk 3]
```diff
@@ -7,13 +7,14 @@ $ORIGIN zone10.com.
...
-              )
+              );
 
   172800 IN NS ns1-03.azure-dns.com.
   172800 IN NS ns2-03.azure-dns.net.
   172800 IN NS ns3-03.azure-dns.org.
   172800 IN NS ns4-03.azure-dns.info.
-;
+
+@ NS ns1-03.azure-dns.com.
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/zone_files/zone10.txt:17]: why is this needed?

**Review by djr-jsr — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py (hunk 1 above)
Comment (WhiteHorseV) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py:594]: In zone1.txt, I have commented out the naptr record types. I will enable them after fixing the zone import logic in another PR.
If I uncomment these lines the unit tests will fail - `DnsParseZoneFiles.test_zone_file_1`

File: src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py (hunk 2 above)
Comment (WhiteHorseV) [reply src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py:212]: I don't think we allow empty value in any other field for any other record type. Otherwise, I guess serialize function call after this tokenize function call will remove that entirely.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/zone_files/zone10.txt (hunk 3 above)
Comment (WhiteHorseV) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/zone_files/zone10.txt]: Accidentally changed it. Reverted back to original file.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py (hunk 1 above)
Comment (WhiteHorseV) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_dns_commands.py]: I removed this entirely and added a separate zone file

File: src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py (hunk 2 above)
Comment (WhiteHorseV) [reply src/azure-cli/azure/cli/command_modules/network/zone_file/parse_zone_file.py:212]: We don't allow empty string for any other record type from azcli. This should be good enough.

**Review by necusjz — APPROVED** [review]
